manifests/params.pp

```diff
@@ -23,121 +22,269 @@
+    'connections.max.idle.ms'                       => '600000',
     'controlled.shutdown.enable'                    => true,
     'controlled.shutdown.max.retries'               => '3',
     'controlled.shutdown.retry.backoff.ms'          => '5000',
-    'auto.leader.rebalance.enable'                  => true,
-    'leader.imbalance.per.broker.percentage'        => '10',
-    'leader.imbalance.check.interval.seconds'       => '300',
-    'offset.metadata.max.bytes'                     => '1024',
+    'controller.socket.timeout.ms'                  => '30000',
+    'default.replication.factor'                    => '1',
+    'fetch.purgatory.purge.interval.requests'       => '1000',
+    'group.max.session.timeout.ms'                  => '30000',
+    'group.min.session.timeout.ms'                  => '6000',
+    'inter.broker.protocol.version'                 => '0.8.2.2',
+    'log.cleaner.backoff.ms'                        => '15000',
+    'log.cleaner.dedupe.buffer.size'                => '134217728',
+    'log.cleaner.delete.retention.ms'               => '86400000',
+    'log.cleaner.enable'                            => true,
+    'log.cleaner.io.buffer.load.factor'             => '0.9',
+    'log.cleaner.io.buffer.size'                    => '524288',
+    'log.cleaner.io.max.bytes.per.second'           => '1.7976931348623157E308',
+    'log.cleaner.min.cleanable.ratio'               => '0.5',
+    'log.cleaner.threads'                           => '1',
+    'log.cleanup.policy'                            => 'delete',
+    'log.index.interval.bytes'                      => '4096',
+    'log.index.size.max.bytes'                      => '10485760',
+    'log.preallocate'                               => false,
+    'log.retention.check.interval.ms'               => '300000',
+    'max.connections.per.ip'                        => '2147483647',
+    'max.connections.per.ip.overrides'              => '',
+    'num.partitions'                                => '1',
+    'principal.builder.class'                       => 'org.apache.kafka.common.security.auth.DefaultPrincipalBuilder',
+    'producer.purgatory.purge.interval.requests'    => '1000',
+    'replica.fetch.backoff.ms'                      => '1000',
+    'reserved.broker.max.id'                        => '1000',
+    'sasl.kerberos.kinit.cmd'                       => '/usr/bin/kinit',
+    'sasl.kerberos.min.time.before.relogin'         => '60000',
+    'sasl.kerberos.principal.to.local.rules'        => '[DEFAULT]',
+    'sasl.kerberos.service.name'                    => '',
+    'sasl.kerberos.ticket.renew.jitter'             => '0.05',
+    'sasl.kerberos.ticket.renew.window.factor'      => '0.8',
+    'security.inter.broker.protocol'                => 'PLAINTEXT',
+    'ssl.cipher.suites'                             => '',
+    'ssl.client.auth'                               => 'none',
+    'ssl.enabled.protocols'                         => '[TLSv1.2, TLSv1.1, TLSv1]',
```

Comment:
I'd say default to TLSv1.2, and they can add more protocols if needed. That's what I did for puppetserver.